execution/gethexec/switch_sequencer.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/arbos"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+)
+
+const (
+	SequencingMode_Espresso    = 0
+	SequencingMode_Centralized = 1
+)
+
+type SwitchSequencer struct {
+	stopwaiter.StopWaiter
+
+	centralized *Sequencer
+	espresso    *EspressoSequencer
+
+	maxHotShotDriftTime time.Duration
+	switchPollInterval  time.Duration
+	lightClient         lightClient.LightClientReaderInterface
+
+	mode int
+}
+
+func NewSwitchSequencer(centralized *Sequencer, espresso *EspressoSequencer, l1client bind.ContractBackend, configFetcher SequencerConfigFetcher) (*SwitchSequencer, error) {
+	config := configFetcher()
+	err := config.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	var lightClient lightClient.LightClientReaderInterface
+	if config.LightClientAddress != "" {
+		lightClient, err = arbos.NewMockLightClientReader(common.HexToAddress(config.LightClientAddress), l1client)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &SwitchSequencer{
+		centralized:         centralized,
+		espresso:            espresso,
+		lightClient:         lightClient,
+		mode:                SequencingMode_Espresso,
+		maxHotShotDriftTime: config.MaxHotShotDriftTime,
+		switchPollInterval:  config.SwitchPollInterval,
+	}, nil
+}
+
+func (s *SwitchSequencer) IsRunningEspressoMode() bool {
+	return s.mode == SequencingMode_Espresso
+}
+
+func (s *SwitchSequencer) SwitchToEspresso(ctx context.Context) error {
+	if s.IsRunningEspressoMode() {
+		return nil
+	}
+	log.Info("Switching to espresso sequencer")
+
+	s.mode = SequencingMode_Espresso
+
+	s.centralized.StopAndWait()
+	return s.espresso.Start(ctx)
+}
+
+func (s *SwitchSequencer) SwitchToCentralized(ctx context.Context) error {
+	if !s.IsRunningEspressoMode() {
+		return nil
+	}
+	s.mode = SequencingMode_Centralized
+	log.Info("Switching to centrialized sequencer")
+
+	s.espresso.StopAndWait()
+	return s.centralized.Start(ctx)
+}
+
+func (s *SwitchSequencer) getRunningSequencer() TransactionPublisher {
+	if s.IsRunningEspressoMode() {
+		return s.espresso
+	}
+	return s.centralized
+}
+
+func (s *SwitchSequencer) PublishTransaction(ctx context.Context, tx *types.Transaction, options *arbitrum_types.ConditionalOptions) error {
+	return s.getRunningSequencer().PublishTransaction(ctx, tx, options)
+}
+
+func (s *SwitchSequencer) CheckHealth(ctx context.Context) error {
+	return s.getRunningSequencer().CheckHealth(ctx)
+}
+
+func (s *SwitchSequencer) Initialize(ctx context.Context) error {
+	err := s.centralized.Initialize(ctx)
+	if err != nil {
+		return err
+	}
+
+	return s.espresso.Initialize(ctx)
+}
+
+func (s *SwitchSequencer) Start(ctx context.Context) error {
+	s.StopWaiter.Start(ctx, s)
+	err := s.getRunningSequencer().Start(ctx)
+	if err != nil {
+		return err
+	}
+
+	if s.lightClient != nil {
```

Comment:
Log a message like "light client not specified, we will only be sequencing mode"